packages/rs-drive/src/state_transition_action/system/partially_use_asset_lock_action/v0/transformer.rs

```diff
@@ -0,0 +1,278 @@
+use crate::state_transition_action::identity::identity_create::v0::IdentityCreateTransitionActionV0;
+use crate::state_transition_action::identity::identity_topup::v0::IdentityTopUpTransitionActionV0;
+use crate::state_transition_action::system::partially_use_asset_lock_action::v0::PartiallyUseAssetLockActionV0;
+use dpp::asset_lock::reduced_asset_lock_value::AssetLockValueGettersV0;
+use dpp::consensus::basic::identity::IdentityAssetLockTransactionOutputNotFoundError;
+use dpp::consensus::ConsensusError;
+use dpp::fee::Credits;
+use dpp::platform_value::Bytes36;
+use dpp::state_transition::identity_create_transition::v0::IdentityCreateTransitionV0;
+use dpp::state_transition::identity_topup_transition::v0::IdentityTopUpTransitionV0;
+
+impl PartiallyUseAssetLockActionV0 {
+    /// try from identity create transition
+    pub fn try_from_identity_create_transition(
+        value: IdentityCreateTransitionV0,
+        asset_lock_initial_balance_amount: Credits,
+        asset_lock_output_script: Vec<u8>,
+        asset_lock_remaining_balance_amount: Credits,
+        desired_used_credits: Credits,
+    ) -> Result<Self, ConsensusError> {
+        let IdentityCreateTransitionV0 {
+            asset_lock_proof,
+            user_fee_increase,
+            ..
+        } = value;
+
+        let asset_lock_outpoint = asset_lock_proof.out_point().ok_or_else(|| {
+            IdentityAssetLockTransactionOutputNotFoundError::new(
+                asset_lock_proof.output_index() as usize
+            )
+        })?;
+
+        let remaining_balance_after_used_credits_are_deducted =
+            asset_lock_remaining_balance_amount.saturating_sub(desired_used_credits);
+
+        let used_credits = std::cmp::min(asset_lock_remaining_balance_amount, desired_used_credits);
```

Comment:
Please add a comment why do we have this logic. (copy to functions bellow)

Reply:
Done